docs/designs/publish.md

```diff
@@ -0,0 +1,43 @@
+# Publish
+
+## Publish Manifest File
+
+`docfx` generates a manifest file for publishing at `{output-path}/.publish.json` after build. It is the entry point for publishing. An example `.publish.json` looks like this:
+
+```javascript
+{
+    "publish": [
+        {
+            "url": "/dotnet/api/system.string",
+            "path": "dotnet/api/system.string.json",
+            "hash": "d41d8cd98f00b204e9800998ecf8427e",
+            "locale": "en-us",
+            "monikers": ["netstandard-2.0"],
+
+            // additional properties needed for publish
+        },
+        {
+            "url": "/dotnet/media/logo.png",
+            "path": "dotnet/media/logo.png",
+            "hash": "d41d8cd98f00b204e9800998ecf8427e",
+            "locale": "en-us",
+            "monikers": ["netstandard-2.0", "netstandard-2.1"],
+
+            // additional properties needed for publish
+        }
```

Comment:
{
            "url": "/{base_path}/media/logo.png", (propose to be: url -> asset_id, but current, we can keep url and publish step can do normalize)
   "path": "{relative_path_to_publish_json_file}"
            "hash": "d41d8cd98f00b204e9800998ecf8427e",
            "monikers": ["netstandard-2.0", "netstandard-2.1"],
   extention_data:{
   
   }
}

Reply:
To clarify `extension_data` means using `[JsonExtensionData]`, the actual JSON is a flattened list.